src/controller/p2p/preheat/controller.go

```diff
@@ -108,8 +109,8 @@ type Controller interface {
 	GetPolicyByName(ctx context.Context, projectID int64, name string) (*policyModels.Schema, error)
 	// UpdatePolicy updates the policy.
 	UpdatePolicy(ctx context.Context, schema *policyModels.Schema, props ...string) error
-	// DeletePolicy deletes the policy by id.
-	DeletePolicy(ctx context.Context, id int64) error
+	// DeletePolicyAndExec deletes the policy by id.
+	DeletePolicyAndExec(ctx context.Context, id int64) error
```

Comment:
I suggest keeping the deletion method as it was and defining an internal func called in the policy deletion method. API handler just calls the deletion method of CTL.